diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/CheckInverseMethodRule.kt

```diff
@@ -0,0 +1,65 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.LPAR
+import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.RPAR
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_ARGUMENT_LIST
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.constants.Warnings.INVERSE_FUNCTION_PREFERRED
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+
+/**
+ * This rule checks if inverse method can be used.
+ * For example if there is !isEmpty() on collection call that it changes it to isNotEmpty()
+ */
+class CheckInverseMethodRule(private val configRules: List<RulesConfig>) : Rule("inverse-method") {
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: EmitType) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == CALL_EXPRESSION && node.text in methodMap.keys) {
+            checkCallExpressionName(node)
+        }
+    }
+
+    private fun checkCallExpressionName(node: ASTNode) {
+        if (node.treeParent.treePrev != null &&
+            node.treeParent.treePrev.elementType == OPERATION_REFERENCE &&
+            node.treeParent.treePrev.text == "!") {
```

Comment:
There could be white spaces or/and comments between `!` and the expression.

Reply:
Fixed